components/edit_post_modal/edit_post_modal.tsx

```diff
@@ -192,13 +246,17 @@ class EditPostModal extends React.PureComponent {
             return;
         }
 
-        if (updatedPost.message === (editingPost.post.message_source || editingPost.post.message)) {
+        if (
+            updatedPost.message ===
+      (editingPost.post?.message_source || editingPost.post?.message)
+        ) {
             // no changes so just close the modal
             this.handleHide();
             return;
         }
 
-        const hasAttachment = editingPost.post.file_ids && editingPost.post.file_ids.length > 0;
+        const hasAttachment =
```

Comment:
We might want to cast this to convert this to boolean so that we don't have to check for `null` or `undefined` here.